### ISSUE TITLE
redirects and auth() -- need more configurable options

### ISSUE THREAD

**Issue by paul-hammant** [rest-assured/rest-assured#948]
Say site example.com/foo responds with 302 for that URL, RestAssured should follow that, sure, even to another-domain.com/foo if that what was instructed.

RestAssured should not use auth() credentials that worked on the first domain (example.com) and pass them to the second domain (another-domain.com).  No browser IRL would do that.

New grammar needed, perhaps:

```
given().
                auth().preemptive().basic("fred", "pw123").forDomain("example.com").
                auth().none().forDomain("another-domain.com").
      when().
                get("/foo")...
```

or

```
given().
                auth().preemptive(redirectsToo()).basic("fred", "pw123").forDomain("example.com").
      when().
                get("/foo")...
```

**Comment by johanhaleby** [comment]
Are you defining the domain/baseUri statically somewhere? Is that why you want to use this?

**Comment by paul-hammant** [comment]
I don't know anything about static domain/baseUris for RestAssured.  I dont really know what your asking.  

The server that's being tested is issuing the 302. That should be a by-default non reuse of the auth credentials for RestAssured (just like browsers).

My app very much like the mvn archetype generated one detailed in http://jooby.org/quickstart/ ... complete with a JUnit test that uses RestAssured to hit an endpoint that 302s

**Comment by johanhaleby** [comment]
Oh ok I get it, so your use case is only applicable for redirects or are there other use cases?

**Comment by paul-hammant** [comment]
Yup, just redirects for me right now and maybe for all the 3xx's. 

It could be that many headers (incl cookies) are n the same class - browsers have a ton of fine-grained rules honed over 20 years:

* Accepts media types when (say) mail.google.com redirects to outlook.com: sure, repeat all those
* Cookies when (say) mail.google.com redirects to outlook.com?, nope none of the ones that were good for google.com should go to outlook.com 

In my case, I had my SUT redirect to an apache domain, and the credentials for my docker HTTP test-bed (user=user, password=password) get forwarded to Apache which blocked the operation because that user didn't exist in the account system, when in reality no basic-auth line would have worked for the operation on Apache. It didn't take me long to work it out (would have been quicker if I was pairing no doubt), and redo the tests so that it didn't matter that RestAssured has this feature, but as I like giving back to excellent pieces of s/w I thought I'd raise this issue :)
